ui/README.md

```diff
@@ -1,76 +1,75 @@
 # Embedded UI
 
-This directory contains the client-side code for cockroach's web admin console.
-These files are embedded into the cockroach binary via the
+This directory contains the client-side code for cockroach's web admin
+console. These files are embedded into the cockroach binary via the
 [go-bindata](https://github.com/jteeuwen/go-bindata) package, which is used to
 generate the `embedded.go` file in this directory.
 
 ## Modification
-If any modifications are made to the contents of this directory, run `make`
-in this directory to generate a new `embedded.go`, which should _then be
+
+If any modifications are made to the contents of this directory, run `make` in
+this directory to generate a new `embedded.go`, which should _then be
 committed along with the original files_. The generated file is committed
-because making our web admin resource requires [additional non-go
-dependencies](#dependencies)
+because compiling our web admin resource requires
+[additional non-go dependencies](#dependencies).
 
 ## Development
-While actively developing the user interface, use `make debug` in order to run
-go-bindata in debug mode.  This will instruct go-bindata to proxy your local
-files (rather than using embedded versions) so you'll be able to edit them live
-without recompiling or restarting the server.
-
-For this change to be picked up, you'll need to run `make build` in the project root.
 
-Note that only those files that were present the last time you ran `make debug`
-will be proxied; if you add a new file, you must run recompile and restart the
-server after running `make debug`.
+While actively developing the user interface, be sure to run the CockroachDB
+server with the environment variable `COCKROACH_DEBUG_UI` set to a truthy
+value, e.g. `COCKROACH_DEBUG_UI=1` (though any value accepted by
+[strconv.ParseBool](https://godoc.org/strconv#ParseBool) will work). This will
+cause the CockroachDB server to serve assets directly from the disk, rather
+than use the compiled-in assets.
 
-Before committing, be sure to run `make` to generate a non-debug version of
-`embedded.go`. This is enforced by our build system, but forgetting to do this
-will result in wasted time waiting for the build.
+In order to regenerate the on-disk assets, be sure to run `make` in this
+directory. This will regenerate `build/app.{css,js}` as well as embedded.go.
 
-## Watch/Livereload
-If you want to automatically recompile/copy the typescript/stylus/index files,
-you can use `make watch`. This runs [Gulp](http://gulpjs.com/) under the hood.
+For faster iteration still, you may want to point your browser at debug.html
+rather than index.html at the web console's top level; this doesn't require
+running `make` at all - instead, all raw assets will be loaded from disk and
+compiled in the browser.
```

Comment:
Had to rewrite a bunch of stuff to address this, PTAL.

On Tue, Jul 5, 2016 at 8:30 PM, mrtracy <notifications@github.com> wrote:

> In ui/README.md
> <https://github.com/cockroachdb/cockroach/pull/7602#discussion_r69660508>:
>
> >
> > -## Watch/Livereload
> > -If you want to automatically recompile/copy the typescript/stylus/index files,
> > -you can use `make watch`. This runs [Gulp](http://gulpjs.com/) under the hood.
> > +For faster iteration still, you may want to point your browser at debug.html
> > +rather than index.html at the web console's top level; this doesn't require
> > +running `make` at all - instead, all raw assets will be loaded from disk and
> > +compiled in the browser.
>
> This seems a bit deceiving, if you've never run make then you don't have
> all the npm/jspm stuff. You do need to run it at least once, and again if
> any dependencies are added or changed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/7602/files/f01c2fadfb2f3518af1f3f7820e1c31a88391a00#r69660508>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABdsPFnJhf9qDEGXYKHqhnXunM-3bnceks5qSvckgaJpZM4JDkyv>
> .
>